schema/spacewalk/postgres/procs/insert_product_vulnerable_packages.sql

```diff
@@ -0,0 +1,60 @@
+--
+-- Copyright (c) 2023 SUSE LLC
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+-- Red Hat trademarks are not licensed under GPLv2. No permission is
+-- granted to use or replicate Red Hat trademarks that are incorporated
+-- in this software or its documentation.
+--
+
+CREATE OR REPLACE PROCEDURE
```

Comment:
Is there any reason to have this logic in PSQL instead of Java? What are the advantages of having i there?
We are trying to have logic in store procedures/functions because it make it much harder to test and debug.

Reply:
The primary reason is performance, this way, we can call procedures in batches. Even with this optimization, writing OVAL data to the database from a ~400 MB file takes 5 minutes on my machine, so it's kind of necessary.

Reply:
Have you tried to use hibernate and flush the inserts after a certain number of inserts (the batch size)? What was the performance degradation on that?
We should consider performance vs maintenance of the code, and to do that we would need to compare performance times for each solution.

Reply:
Yes, I tried Hibernate while flushing after inserting a batch and the performance was horrible, even with native queries. See [this](https://github.com/uyuni-project/uyuni-rfc/pull/80#discussion_r1257750349) for more details.